hil/src/ssh_wrapper.rs

```diff
@@ -0,0 +1,127 @@
+use color_eyre::{eyre::bail, Result};
+use ssh2::Session as Ssh2Session;
+use std::io::Read;
+use std::net::TcpStream;
+use tracing::{debug, info};
+
+/// SSH wrapper that supports both password and key authentication
+/// Will be removed in favor of ssh keys
+pub struct SshWrapper {
+    session: Ssh2Session,
+}
+
+impl SshWrapper {
+    pub async fn connect(
+        host: String,
+        port: u16,
+        username: String,
+        password: String,
```

Comment:
given that you use this same set of arguments in several functions, maybe it makes sense to create a simple SshConnectArgs struct.